### PR TITLE
Issue 78: Summary Statistics How-to Guide

### LINKED ISSUE #78: docs: Sample data at regular intervals How-to-Guide
- [x]  Show continuous stats computations with Agents i.e moving averages,  min, max etc.
- [ ]  Store raw data in a `rawData` map lane for a time window say 1 minute
- [ ]  Show sampling by moving the stats to a `stats` map lane with timestamps after the time window and start storing the raw data for the current window in the `rawData` map lane

Will be nice to demonstrate this with multiple Web Agents and show cli commands to stream data from a couple of Agents

### PR DESCRIPTION
**Pull request by brohitbrose**
Fixes #78.

### CODE REVIEW

File: src/_guides/time-series.md [hunk 1]
```diff
@@ -213,3 +214,46 @@ We simply check for this and call `trimHistory()` if the number calculated is le
...
+```
+
+respectively.
+
+### Chained Values
+
+Using compound keys precludes using `MapLane.get(timestamp)` to quickly seek entries under a provided timestamp.
+If your application requires this _uncommon_ functionality, and _only if_ iterating over the compound-keyed `MapLane` proves too inefficient, implement the timeseries as a `MapLane<Long, List<Value>>` (or similar) instead.
+
+_Note: this representation makes it trickier to both iterate over and join to the `MapLane`._
```
Comment (brohitbrose) [inline src/_guides/time-series.md:259]: I want to include a link in the word `join`. Should this link to the aggregations how-to guide or the `JoinMapLane` reference?

**Comment by brohitbrose** [comment]
Have drafted this PR. I plan to add `swim-cli` commands across all the how-to guides and believe I should be able to pull it off in one shot shortly to soon. Will update when this is done.

File: src/_guides/summary-statistics.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+A few things to note:
+
+- Running the `Main` class runs both a Swim server _and_ a simulator that feeds with data.
+- `src/main/resources/server.recon` defines routing that activates `TowerAgent`, `BucketedTowerAgent`, and `WindowedTowerAgent` all at once to facilitate exploration; in a real application, you would likely pick a single implementation.
+
+The following `swim-cli` commands are available while the process runs:
+
+- `swim-cli sync -h warp://localhost:9001 -n /tower/$ID -l summary` to stream the `summary` lane's entry from within a `TowerAgent` instance
+- `swim-cli sync -h warp://localhost:9001 -n /bucketed/$ID -l summaries` to stream the `summaries` lane's entries from within a `BucketedTowerAgent` instance
+- `swim-cli sync -h warp://localhost:9001 -n /windowed/$ID -l summaries` to stream the `summaries` lane's entries from within a `WindowedTowerAgent` instance
```
Comment (brohitbrose) [inline src/_guides/summary-statistics.md:298]: Is this formatting hard on the eyes when viewed directly on the site?

File: src/_guides/time-series.md [hunk 3]
```diff
@@ -213,3 +214,46 @@ We simply check for this and call `trimHistory()` if the number calculated is le
...
+key = Record.create(2).slot("timestamp", timestamp).slot("id", id);
+```
+
+and
+
+```java
+key = Record.create(2).item(timestamp).item(id);
+```
+
+`MapLanes` are always ordered by key, and `Value`s are compared sequentially in field order (similarly to how _composite keys_ work in SQL).
```
Comment (brohitbrose) [inline src/_guides/time-series.md:238]: Is our convention `MapLanes` or `MapLane`s? Most of the articles use the latter, but it looks jarring when rendered. If it were up to me I would do the former (fixing it in all the other articles too).

**Comment by brohitbrose** [comment]
One final thing, I did not exactly follow the guidelines suggested by @ajay-gov in #78 when creating this PR. I found that it overlapped too much with the Time Series guide. I did include a _Note_ detailing how to accomplish those tasks. Happy to revisit based on feedback.

**Comment by ajay-gov** [comment]
Looks good for the most part. Couple of comments, some of them could be just me
- Can we change `mean_ul_sinr` and the `rrc_re_establishment_failures` to simpler names. They are carryovers from Verizon data, might be good to avoid them Maybe `bandwidth` and `numDisconnects`?
-  Having uri paths as `bucketed`and `windowed` might send  the wrong message since they typically represent entities and not behaviors. Maybe replace with `towerB` and `towerW` or  something like that
-  Can we document `Windowed` first and then do `Bucketed` next  since that seems to be in the order of difficulty. We can always explain that `Windowed` doesn't handle out-of-order and how `Bucketed` will help
-  An explanation that `timestamp / SAMPLE_PERIOD_MS * SAMPLE_PERIOD_MS` results in a bucket that represents the start of a minute in milliseconds will be useful here with an example i.e timestamp with `x' representing 15:34:33 will become 15:34:00 in milliseconds. Otherwise it's a bit cryptic, any thoughts?
- Maybe show some cli output in the doc? 

I do find the variance computation and using the delta for the mean computation pretty cool but is that a bit too complicated?

**Comment by brohitbrose** [comment]
1/2: agreed, will change the names

3: Yeah it's interesting, I find bucketing to be the simpler implementation, though trickier conceptually. Windowing into bucketing may flow better, will give it a try.

4/5: Will do, good idea

I'm leaning toward keeping the variance for now; it's mathematically more complicated but it serves as encouragement for people to seek out streaming algorithms since it demonstrates that non-obvious problems often have solutions.

Thanks for the review!

**Comment by ajay-gov** [comment]
Yeah let's keep the variance, it does add some weight to the content. Using the POJO for it  makes the implementation quite nice actually and unit testable.

**Comment by brohitbrose** [comment]
I was unable to move Windowing before Bucketing and have the flow feel natural. Have accounted for all of the other suggestions, though.

**Comment by ajay-gov** [comment]
> I was unable to move Windowing before Bucketing and have the flow feel natural. Have accounted for all of the other suggestions, though.

No worries, all good and reads quite well.

**Review by ajay-gov — APPROVED** [review]

**Review by afolson — APPROVED** [review]

**Pull request merged**